src/handler.ts

```diff
@@ -142,15 +154,19 @@ const jumpTable: JumpTable = {
         return sql
       }
     },
-    Delete: (resourceId: string, props: RoleProps) => {
-      // TODO: if user is owner of a database, assign ownership to master user?
+    Delete: (resourceId: string, _props: RoleProps) => {
       return [
         "start transaction",
         format(
-          "DO $$BEGIN\nIF EXISTS (select from pg_catalog.pg_roles WHERE rolname = '%s') AND EXISTS (select from pg_database WHERE datname = '%s') THEN revoke all privileges on database %I from %I; END IF;\nEND$$;",
+          `DO $$
+BEGIN
+  IF EXISTS (select from pg_catalog.pg_roles WHERE rolname = '%s') THEN
+    reassign OWNED by %I to posgres;
```

Comment:
You have a hard-coded role here, which people may not have. Also from that discussion reassigning can lead to privilege escalation, so not keen on that approach. Can you remove this from the PR? I can merge most of it, but not this bit.

Reply:
Will do.

Reply:
From [that thread](https://stackoverflow.com/posts/54527768/revisions), I suggest you add something like this:

```sql
DO $$ 
DECLARE
    r RECORD;
BEGIN
    FOR r IN (SELECT schemaname FROM pg_schemas) LOOP
      EXECUTE 'REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA ' || quote_ident(r.schemaname) || ' FROM %s';
      EXECUTE 'REVOKE ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA ' || quote_ident(r.schemaname) || ' FROM %s';
      EXECUTE 'REVOKE ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA ' || quote_ident(r.schemaname) || ' FROM %s';
      EXECUTE 'REVOKE ALL ON SCHEMA ' || quote_ident(r.schemaname) || ' FROM %I';
    END LOOP;
END $$;
```

Typing this from memory, and needs testing.

And you could add:

```sql
REVOKE CONNECT ON DATABASE db_name FROM role_name;
```

Keep the drop OWNED by %I as the first statement.

Reply:
But this all needs a bit of work, I prefer it if you move this to a separate PR.

Reply:
Sequences specifically is what cause my failure but I can work that into a seperate PR